lib/grack/server.rb

```diff
@@ -43,13 +37,11 @@ def _call(env)
       @req = Rack::Request.new(env)
 
       cmd, path, @reqfile, @rpc = match_routing
-
       return render_method_not_allowed if cmd == 'not_allowed'
       return render_not_found if !cmd
 
       @dir = get_git_dir(path)
```

Comment:
Is it feasible to not use `@dir` and just `git.repo` instead? `get_git_dir` can then just be `get_git` and return the `Git::Grack` instance. I'd also use the `@git.valid_git_dir?` (or `@git.valid_repo?`) on the next line here instead of inside `Grack::Git`, that way the behavior is a bit more obvious then `!@dir`.